[EGL] Merge GLContext and GLContextEGL

carlosgcampos opened the pull request:
#### e322dc6f4a1493e12924c3d4e8f51351ed00a733
<pre>
[EGL] Merge GLContext and GLContextEGL
<a href="https://bugs.webkit.org/show_bug.cgi?id=254270">https://bugs.webkit.org/show_bug.cgi?id=254270</a>

Reviewed by Adrian Perez de Castro.

GLContextEGL is now the only implementation of GLContext, so there&apos;s no
reason to have a base abstract class.

* Source/WebCore/Headers.cmake:
* Source/WebCore/PlatformPlayStation.cmake:
* Source/WebCore/PlatformWin.cmake:
* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/SourcesWPE.txt:
* Source/WebCore/platform/graphics/GLContext.cpp: Removed.
* Source/WebCore/platform/graphics/GLContext.h: Removed.
* Source/WebCore/platform/graphics/PlatformDisplay.cpp:
(WebCore::PlatformDisplay::sharingGLContext):
(WebCore::PlatformDisplay::initializeEGLDisplay):
* Source/WebCore/platform/graphics/egl/GLContext.cpp: Renamed from Source/WebCore/platform/graphics/egl/GLContextEGL.cpp.
(WebCore::initializeOpenGLShimsIfNeeded):
(WebCore::GLContext::errorString):
(WebCore::GLContext::lastErrorString):
(WebCore::GLContext::getEGLConfig):
(WebCore::GLContext::createWindowContext):
(WebCore::GLContext::createPbufferContext):
(WebCore::GLContext::createSurfacelessContext):
(WebCore::GLContext::createContext):
(WebCore::GLContext::createOffscreen):
(WebCore::GLContext::createSharing):
(WebCore::GLContext::GLContext):
(WebCore::GLContext::~GLContext):
(WebCore::GLContext::createContextForEGLVersion):
(WebCore::GLContext::makeContextCurrent):
(WebCore::GLContext::current):
(WebCore::GLContext::swapBuffers):
(WebCore::GLContext::platformContext):
(WebCore::GLContext::isExtensionSupported):
(WebCore::GLContext::version):
* Source/WebCore/platform/graphics/egl/GLContext.h: Added.
(WebCore::GLContext::display const):
(WebCore::GLContext::config const):
* Source/WebCore/platform/graphics/egl/GLContextEGL.h: Removed.
* Source/WebCore/platform/graphics/egl/GLContextEGLLibWPE.cpp:
(WebCore::GLContext::GLContext):
(WebCore::GLContext::createWindowSurfaceWPE):
(WebCore::GLContext::createWPEContext):
(WebCore::GLContext::destroyWPETarget):
(WebCore::GLContextEGL::GLContextEGL): Deleted.
(WebCore::GLContextEGL::createWindowSurfaceWPE): Deleted.
(WebCore::GLContextEGL::createWPEContext): Deleted.
(WebCore::GLContextEGL::destroyWPETarget): Deleted.
* Source/WebCore/platform/graphics/egl/GLContextEGLWayland.cpp:
(WebCore::GLContext::GLContext):
(WebCore::GLContext::createWindowSurfaceWayland):
(WebCore::GLContext::createWaylandContext):
(WebCore::GLContext::destroyWaylandWindow):
(WebCore::GLContextEGL::GLContextEGL): Deleted.
(WebCore::GLContextEGL::createWindowSurfaceWayland): Deleted.
(WebCore::GLContextEGL::createWaylandContext): Deleted.
(WebCore::GLContextEGL::destroyWaylandWindow): Deleted.
* Source/WebCore/platform/graphics/egl/GLContextEGLX11.cpp:
(WebCore::GLContext::GLContext):
(WebCore::GLContext::createWindowSurfaceX11):
(WebCore::GLContext::createPixmapContext):
(WebCore::GLContextEGL::GLContextEGL): Deleted.
(WebCore::GLContextEGL::createWindowSurfaceX11): Deleted.
(WebCore::GLContextEGL::createPixmapContext): Deleted.
* Source/WebCore/platform/graphics/gstreamer/PlatformDisplayGStreamer.cpp:
* Source/WebCore/platform/graphics/libwpe/PlatformDisplayLibWPE.cpp:
(WebCore::PlatformDisplayLibWPE::initialize):
* Source/WebCore/platform/graphics/texmap/TextureMapperPlatformLayerDmabuf.cpp:
(WebCore::TextureMapperPlatformLayerDmabuf::validateTexture):
* Source/WebCore/platform/graphics/wayland/PlatformDisplayWayland.cpp:
* Source/WebCore/platform/xr/openxr/PlatformXROpenXR.cpp:
(PlatformXR::OpenXRDevice::initializeTrackingAndRendering):
* Source/WebCore/platform/xr/openxr/PlatformXROpenXR.h:
* Source/WebKit/GPUProcess/graphics/wc/WCSceneContext.cpp:
(WebKit::WCSceneContext::WCSceneContext):
* Source/WebKit/Shared/CoordinatedGraphics/threadedcompositor/ThreadedCompositor.cpp:
(WebKit::ThreadedCompositor::createGLContext):
* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
(WebKit::WebKitProtocolHandler::handleGPU):
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreWayland.cpp:
(WebKit::tryInitializeEGL):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHostTextureMapper.cpp:
(WebKit::LayerTreeHost::LayerTreeHost):

Canonical link: <a href="https://commits.webkit.org/262074@main">https://commits.webkit.org/262074@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/74aa5312edf0d917c8ed6bdec9ae47de408b1350

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/294 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/309 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/322 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/299 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/277 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/294 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/393 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/473 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/529 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/299 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/330 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/291 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/324 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/362 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/284 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/356 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/259 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/300 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/282 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/383 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/252 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/293 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/116 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/286 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebCore/platform/graphics/egl/GLContext.cpp:
```diff
@@ -52,10 +49,33 @@
+
+static bool initializeOpenGLShimsIfNeeded()
+{
+#if USE(OPENGL_ES) || USE(LIBEPOXY) || (USE(ANGLE) && !(PLATFORM(GTK) || PLATFORM(WPE)))
```
`USE(LIBEPOXY)` is guaranteed to be true for both GTK and WPE, so this can be simplified.

carlosgcampos replied:
Ok

mcatanzaro reviewed (commented):
I think you should rename GLContextEGLLibWPE.cpp, GLContextEGLWayland.cpp, and GLContextEGLX11.cpp as well.

carlosgcampos committed:
[EGL] Merge GLContext and GLContextEGL
https://bugs.webkit.org/show_bug.cgi?id=254270

Reviewed by Adrian Perez de Castro.

GLContextEGL is now the only implementation of GLContext, so there's no
reason to have a base abstract class.

* Source/WebCore/Headers.cmake:
* Source/WebCore/PlatformPlayStation.cmake:
* Source/WebCore/PlatformWin.cmake:
* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/SourcesWPE.txt:
* Source/WebCore/platform/graphics/GLContext.cpp: Removed.
* Source/WebCore/platform/graphics/GLContext.h: Removed.
* Source/WebCore/platform/graphics/PlatformDisplay.cpp:
(WebCore::PlatformDisplay::sharingGLContext):
(WebCore::PlatformDisplay::initializeEGLDisplay):
* Source/WebCore/platform/graphics/egl/GLContext.cpp: Renamed from Source/WebCore/platform/graphics/egl/GLContextEGL.cpp.
(WebCore::initializeOpenGLShimsIfNeeded):
(WebCore::GLContext::errorString):
(WebCore::GLContext::lastErrorString):
(WebCore::GLContext::getEGLConfig):
(WebCore::GLContext::createWindowContext):
(WebCore::GLContext::createPbufferContext):
(WebCore::GLContext::createSurfacelessContext):
(WebCore::GLContext::createContext):
(WebCore::GLContext::createOffscreen):
(WebCore::GLContext::createSharing):
(WebCore::GLContext::GLContext):
(WebCore::GLContext::~GLContext):
(WebCore::GLContext::createContextForEGLVersion):
(WebCore::GLContext::makeContextCurrent):
(WebCore::GLContext::current):
(WebCore::GLContext::swapBuffers):
(WebCore::GLContext::platformContext):
(WebCore::GLContext::isExtensionSupported):
(WebCore::GLContext::version):
* Source/WebCore/platform/graphics/egl/GLContext.h: Added.
(WebCore::GLContext::display const):
(WebCore::GLContext::config const):
* Source/WebCore/platform/graphics/egl/GLContextEGL.h: Removed.
* Source/WebCore/platform/graphics/egl/GLContextEGLLibWPE.cpp:
(WebCore::GLContext::GLContext):
(WebCore::GLContext::createWindowSurfaceWPE):
(WebCore::GLContext::createWPEContext):
(WebCore::GLContext::destroyWPETarget):
(WebCore::GLContextEGL::GLContextEGL): Deleted.
(WebCore::GLContextEGL::createWindowSurfaceWPE): Deleted.
(WebCore::GLContextEGL::createWPEContext): Deleted.
(WebCore::GLContextEGL::destroyWPETarget): Deleted.
* Source/WebCore/platform/graphics/egl/GLContextEGLWayland.cpp:
(WebCore::GLContext::GLContext):
(WebCore::GLContext::createWindowSurfaceWayland):
(WebCore::GLContext::createWaylandContext):
(WebCore::GLContext::destroyWaylandWindow):
(WebCore::GLContextEGL::GLContextEGL): Deleted.
(WebCore::GLContextEGL::createWindowSurfaceWayland): Deleted.
(WebCore::GLContextEGL::createWaylandContext): Deleted.
(WebCore::GLContextEGL::destroyWaylandWindow): Deleted.
* Source/WebCore/platform/graphics/egl/GLContextEGLX11.cpp:
(WebCore::GLContext::GLContext):
(WebCore::GLContext::createWindowSurfaceX11):
(WebCore::GLContext::createPixmapContext):
(WebCore::GLContextEGL::GLContextEGL): Deleted.
(WebCore::GLContextEGL::createWindowSurfaceX11): Deleted.
(WebCore::GLContextEGL::createPixmapContext): Deleted.
* Source/WebCore/platform/graphics/gstreamer/PlatformDisplayGStreamer.cpp:
* Source/WebCore/platform/graphics/libwpe/PlatformDisplayLibWPE.cpp:
(WebCore::PlatformDisplayLibWPE::initialize):
* Source/WebCore/platform/graphics/texmap/TextureMapperPlatformLayerDmabuf.cpp:
(WebCore::TextureMapperPlatformLayerDmabuf::validateTexture):
* Source/WebCore/platform/graphics/wayland/PlatformDisplayWayland.cpp:
* Source/WebCore/platform/xr/openxr/PlatformXROpenXR.cpp:
(PlatformXR::OpenXRDevice::initializeTrackingAndRendering):
* Source/WebCore/platform/xr/openxr/PlatformXROpenXR.h:
* Source/WebKit/GPUProcess/graphics/wc/WCSceneContext.cpp:
(WebKit::WCSceneContext::WCSceneContext):
* Source/WebKit/Shared/CoordinatedGraphics/threadedcompositor/ThreadedCompositor.cpp:
(WebKit::ThreadedCompositor::createGLContext):
* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
(WebKit::WebKitProtocolHandler::handleGPU):
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreWayland.cpp:
(WebKit::tryInitializeEGL):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHostTextureMapper.cpp:
(WebKit::LayerTreeHost::LayerTreeHost):

Canonical link: https://commits.webkit.org/262074@main
